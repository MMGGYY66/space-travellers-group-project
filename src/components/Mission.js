import React from 'react';

const Mission = () => (
  <h1>Missions List</h1>
);

export default Mission;
