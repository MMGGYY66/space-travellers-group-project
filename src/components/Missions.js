import React from 'react';

const Missions = () => (
  <h1>Missions List</h1>
);

export default Missions;
