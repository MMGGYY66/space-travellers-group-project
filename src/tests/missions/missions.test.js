import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { Provider } from 'react-redux';
import { configureStore } from '@reduxjs/toolkit';
import Missions from '../../pages/Missions';
import Mission from '../../components/Mission';
import missionsReducer from '../../Redux/Missions/missions';

const initialMissionState = {
  missions: [
    {
      id: 'missionID',
      name: 'missionName',
      description: 'missionDescription',
      reserved: false,
    },
  ],
};

const store = configureStore({
  reducer: missionsReducer,
  preloadedState: initialMissionState,
});

describe('Missions', () => {
  afterEach(cleanup);

  it('Should render a single mission', () => {
    const { mission } = render(
      <Provider store={store}>
        <Mission mission={initialMissionState.missions[0]} />
      </Provider>,
    );
    expect(mission).toMatchSnapshot();
  });

  it('Should render the Missions page correctly', () => {
    const { missions } = render(
      <Provider store={store}>
        <Missions />
      </Provider>,
    );
    expect(missions).toMatchSnapshot();
  });

  it('Should display the mission page with correct information', () => {
    const { getByText } = render(
      <>
        <Provider store={store}>
          <Missions />
        </Provider>
        ,
      </>,
    );
    const missionName = getByText('missionName');
    expect(missionName).toBeInTheDocument();
  });
});
