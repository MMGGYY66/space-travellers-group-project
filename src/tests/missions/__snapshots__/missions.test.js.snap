// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Missions Should render a single mission 1`] = `undefined`;

exports[`Missions Should render the Missions page correctly 1`] = `undefined`;
